src/mds/MDSRank.cc

```diff
@@ -3579,6 +3584,32 @@ void MDSRank::command_cache_drop(uint64_t timeout, Formatter *f, Context *on_fin
   request->send();
 }
 
+int MDSRank::command_dump_rstats(Formatter *f, std::ostream &ss)
+{
+  if (get_nodeid() != 0) {
+    ss << "I am not mds.0";
+    return -EINVAL;
+  }
+
+  CInode *root = mdcache->get_root();
+  if (!root) {
+    ss << "root inode is not in cache";
+    return -EINVAL;
+  }
+
+  nest_info_t rstat = root->get_projected_inode()->rstat;
+  f->open_object_section("root rstats");
+  f->dump_unsigned("rbytes", rstat.rbytes);
+  f->dump_unsigned("rfiles", rstat.rfiles);
+  f->dump_unsigned("rsubdirs", rstat.rsubdirs);
+  f->dump_unsigned("rsnaps", rstat.rsnaps);
+  f->dump_stream("rctime") << rstat.rctime;
+  f->dump_stream("flushed_to") << locker->get_rstat_flushed_to();
+  f->dump_stream("mds_time") << ceph_clock_now();
+  f->close_section();
```

Comment:
```suggestion
  f->open_object_section("root rstats");
  f->dump_object("rstats", rstat);
  f->dump_string("path", path);
  f->dump_stream("flushed_to") << locker->get_rstat_flushed_to();
  f->dump_stream("mds_time") << ceph_clock_now();
  f->close_section();
```

?

Reply:
this will include 'dirty_from' field of rstat, which is meaningless for root inode